homeassistant/components/alert/__init__.py

```diff
@@ -311,3 +316,32 @@ async def async_toggle(self, **kwargs: Any) -> None:
         if self._ack:
             return await self.async_turn_on()
         return await self.async_turn_off()
+
+    async def async_update_config(self, config: ConfigType) -> None:
+        """Handle when the config is updated."""
+        self._attr_name = config[CONF_NAME]
+        self._alert_state = config[CONF_STATE]
+        self._skip_first = config[CONF_SKIP_FIRST]
+        self._data = config.get(CONF_DATA)
+        self._watched_entity_id = config[CONF_ENTITY_ID]
+        self._message_template = config.get(CONF_ALERT_MESSAGE)
+        self._done_message_template = config.get(CONF_DONE_MESSAGE)
+        self._title_template = config.get(CONF_TITLE)
+        self._notifiers = config[CONF_NOTIFIERS]
+        self._can_ack = config[CONF_CAN_ACK]
+        self._delay = [timedelta(minutes=val) for val in config[CONF_REPEAT]]
```

Comment:
This is the same code as in `__init__`, can it be moved to a helper?

Reply:
Moved to `_update_from_config` helper function.